pkg/store/bucket.go

```diff
@@ -220,6 +220,10 @@ type BucketStore struct {
 	advLabelSets             []storepb.LabelSet
 	enableCompatibilityLabel bool
 	enableIndexHeader        bool
+
+	// Reencode postings using diff+varint+snappy when storing to cache.
+	// This makes them smaller, but takes extra CPU and memory.
```

Comment:
`memory` is tricky here. Because if you in-mem cache it might reduce overall memory a lot.

Reply:
Added extra `// When used with in-memory cache, memory usage should decrease overall, thanks to postings being smaller.` sentence to the comment.

Reply:
Thanks!